test/extended/jenkins/plugin.go

```diff
@@ -203,11 +460,269 @@ var _ = g.Describe("[jenkins][Slow] openshift pipeline plugin", func() {
+			ginkgolog("\n\nJenkins logs>\n%s\n\n", logs)
 			o.Expect(err).NotTo(o.HaveOccurred())
 
 		})
 
+		g.It("jenkins-plugin test trigger build with envs", func() {
+
+			jobName := "test-build-with-env-job"
+			data := j.readJenkinsJob("build-with-env-job.xml", oc.Namespace())
+			j.createItem(jobName, data)
+			jmon := j.startJob(jobName)
+			jmon.await(3 * time.Minute)
+
+			logs, err := j.getLastJobConsoleLogs(jobName)
+			ginkgolog("\n\nJenkins logs>\n%s\n\n", logs)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			// the build and deployment is by far the most time consuming portion of the test jenkins job;
+			// we leverage some of the openshift utilities for waiting for the deployment before we poll
+			// jenkins for the successful job completion
+			g.By("waiting for frontend deployments as signs that the build has finished")
+			err = exutil.WaitForADeploymentToComplete(oc.KubeREST().ReplicationControllers(oc.Namespace()), "frontend", oc)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			g.By("get build console logs and see if succeeded")
+			_, err = j.waitForContent("Finished: SUCCESS", 200, 3*time.Minute, "job/%s/lastBuild/consoleText", jobName)
+
+			assertEnvVars(oc, "frontend-", map[string]string{
+				"a": "b",
+				"C": "D",
+				"e": "",
+			})
+
+		})
+
+		g.It("jenkins-plugin test trigger build DSL", func() {
+
+			buildsBefore, err := oc.REST().Builds(oc.Namespace()).List(kapi.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			data, err := j.buildDSLJob(oc.Namespace(),
+				"node{",
+				"openshiftBuild( namespace:'PROJECT_NAME', bldCfg: 'frontend', env: [ [ name : 'a', value : 'b' ], [ name : 'C', value : 'D' ], [ name : 'e', value : '' ] ] )",
+				"}",
+			)
+
+			jobName := "test-build-dsl-job"
+			j.createItem(jobName, data)
+			monitor := j.startJob(jobName)
+			err = monitor.await(3 * time.Minute)
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = wait.Poll(1*time.Second, 3*time.Minute, func() (bool, error) {
+				buildsAfter, err := oc.REST().Builds(oc.Namespace()).List(kapi.ListOptions{})
+				o.Expect(err).NotTo(o.HaveOccurred())
+				return (len(buildsAfter.Items) != len(buildsBefore.Items)), nil
+			})
+
+			buildsAfter, err := oc.REST().Builds(oc.Namespace()).List(kapi.ListOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+			o.Expect(len(buildsAfter.Items)).To(o.Equal(len(buildsBefore.Items) + 1))
+
+			log, err := j.getLastJobConsoleLogs(jobName)
+			ginkgolog("Job logs>>\n%s\n\n", log)
+
+			assertEnvVars(oc, "frontend-", map[string]string{
+				"a": "b",
+				"C": "D",
+				"e": "",
+			})
+
+		})
+
+		g.It("jenkins-plugin test exec DSL", func() {
+
+			// Create a running pod in which we can execute our commands
+			oc.Run("new-app").Args("https://github.com/openshift/ruby-hello-world").Execute()
+
+			var targetPod *kapi.Pod
+			err := wait.Poll(time.Second, 3*time.Minute, func() (bool, error) {
```

Comment:
@jupierce I ran this test locally and for whatever reason the 3 minute timeout wasn't long enough ... the build hadn't finished pushing the image when it timed out.